curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentNode.java

```diff
@@ -72,9 +72,11 @@
     private final long ttl;
     private final AtomicReference<byte[]> data = new AtomicReference<byte[]>();
     private final AtomicReference<State> state = new AtomicReference<State>(State.LATENT);
-    private final AtomicBoolean authFailure = new AtomicBoolean(false);
+    private volatile boolean authFailure;
+    private volatile boolean parentCreationFailure;
     private final BackgroundCallback backgroundCallback;
     private final boolean useProtection;
+    private final boolean useParentCreation;
```

Comment:
why not name it the same as `creatingParentContainersIfNeeded`?

Reply:
compatible with `useProtection` seems reasonable...though I think we make mistake at the first time.

Reply:
Yeah I wanted to be consistent with the other naming, but I'm not opposed to named it `creatingParentContainersIfNeeded`. But what `useProtection` should be?